### PR TITLE
Mgmt: make createAsync return Mono<ModelT>

### PR DESCRIPTION
**Pull request by ChenTanyi**
close https://github.com/Azure/azure-sdk-for-java/pull/15074 
we don't need invoke order anyone, only add some comment in TaskGroup

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/RoleAssignmentHelper.java [hunk 1]
```diff
@@ -94,7 +94,7 @@ public RoleAssignmentHelper withAccessTo(final String scope, final BuiltInRole a
                     .withBuiltInRole(asRole)
                     .withScope(resourceScope)
                     .createAsync()
-                    .last()
+                    .cast(Indexable.class)
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/RoleAssignmentHelper.java:97]: Why need a `cast`? Does it need `Flux<Indexable>`?

File: sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/ServicePrincipalImpl.java [hunk 2]
```diff
@@ -199,7 +199,7 @@ public String applicationId() {
                                 .createAsync())
                     .doOnNext(
                         indexable ->
-                            cachedRoleAssignments.put(((RoleAssignment) indexable).id(), (RoleAssignment) indexable))
+                            cachedRoleAssignments.put((indexable).id(), indexable))
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/ServicePrincipalImpl.java:202]: Seems just `indexable.id()`?

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachinePopularImageTests.java [hunk 3]
```diff
@@ -44,7 +45,7 @@ public void canCreateAllPopularImageVM() {
         }
 
         for (KnownLinuxVirtualMachineImage image : KnownLinuxVirtualMachineImage.values()) {
-            Flux<Indexable> flux = computeManager.virtualMachines()
+            Mono<VirtualMachine> flux = computeManager.virtualMachines()
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachinePopularImageTests.java:48]: If it is `Mono`, do not name it `flux`.

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachinePopularImageTests.java:48]: Same for `vmFluxes`.

File: sdk/resourcemanager/azure-resourcemanager-samples/src/main/java/com/azure/resourcemanager/appservice/samples/ManageWebAppSourceControlAsync.java [hunk 4]
```diff
@@ -70,9 +70,9 @@ public static boolean runSample(final Azure azure) {
                     .withJavaVersion(JavaVersion.JAVA_8_NEWEST)
                     .withWebContainer(WebContainer.TOMCAT_8_0_NEWEST)
                     .createAsync()
-                    .flatMap(indexable -> {
+                    .flatMapMany(indexable -> {
                         if (indexable instanceof WebApp) {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-samples/src/main/java/com/azure/resourcemanager/appservice/samples/ManageWebAppSourceControlAsync.java:74]: Likely this `if` is no longer needed?

**Review by weidongxu-microsoft — APPROVED** [review]
Are we able to delete `Utils.rootResource`?

File: sdk/resourcemanager/azure-resourcemanager-samples/src/main/java/com/azure/resourcemanager/appservice/samples/ManageWebAppSourceControlAsync.java (hunk 4 above)
Comment (ChenTanyi) [reply sdk/resourcemanager/azure-resourcemanager-samples/src/main/java/com/azure/resourcemanager/appservice/samples/ManageWebAppSourceControlAsync.java:74]: Yes, there are lots of `if` not needed, but I don't want to change so much.

File: sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/RoleAssignmentHelper.java (hunk 1 above)
Comment (ChenTanyi) [reply sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/RoleAssignmentHelper.java:97]: Some are `dependency` or `post dependency`, some are in test for `Flux`

File: sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/ServicePrincipalImpl.java (hunk 2 above)
Comment (ChenTanyi) [reply sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/ServicePrincipalImpl.java]: done

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachinePopularImageTests.java (hunk 3 above)
Comment (ChenTanyi) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachinePopularImageTests.java]: done

File: sdk/resourcemanager/azure-resourcemanager-samples/src/main/java/com/azure/resourcemanager/appservice/samples/ManageWebAppSourceControlAsync.java (hunk 4 above)
Comment (ChenTanyi) [reply sdk/resourcemanager/azure-resourcemanager-samples/src/main/java/com/azure/resourcemanager/appservice/samples/ManageWebAppSourceControlAsync.java]: fixed, it cause spotbugs error

**Review by yungezz — APPROVED** [review]

**Pull request merged**